### DISCUSSION TITLE
Zod for Validation

### DISCUSSION THREAD

**Discussion by ChristoRibeiro** [instantcommerce/next-api-decorators#480] [Q&A] [answered]
In my projects I use [Zod](https://zod.dev) a lot. @ggurkal do you think it's possible/easy to use Zod for Validation?

**Comment by ggurkal** [comment, marked as answer]
Hey @ChristoRibeiro

It sure is possible. The only thing you need is a validation pipe that uses zod (the original one is [here](https://github.com/instantcommerce/next-api-decorators/blob/master/lib/pipes/validators/validation.pipe.ts)), or you can take some inspiration from https://github.com/kbkk/abitia/tree/master/packages/zod-dto how to do it with DTOs.

**Reply by ChristoRibeiro** [reply]
Nice! I will give a try soon. 

I will keep you in touch.

**Reply by Yhozen** [reply]
Hey! I gave it a try and it worked, see #509
